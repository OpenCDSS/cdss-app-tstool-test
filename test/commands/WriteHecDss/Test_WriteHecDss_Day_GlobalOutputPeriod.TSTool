# Test writing a couple of HEC-DSS time series to a file when the global output period is specified.
#@os Windows
StartLog(LogFile="Results/Test_WriteHecDss_Day_GlobalOutputPeriod.TSTool.log")
# SetDebugLevel(LogFileLevel=10)
# Write a single blank time series to the dss file to allow WriteHecDss to be closed; allows deleting below.
NewTimeSeries(Alias="Blank",NewTSID="Blank...Month",SetStart="1960-01",SetEnd="1960-12",Units="CFS")
WriteHecDss(OutputFile="Results\Test_WriteHecDss_Day_GlobalOutputPeriod_out.dss",Type=PER-AVER,Close=True)
Free(TSList=AllMatchingTSID,TSID="*")
RemoveFile(InputFile="Results\Test_WriteHecDss_Day_GlobalOutputPeriod_out.dss")
RemoveFile(InputFile="Results\Test_WriteHecDss_Day_GlobalOutputPeriod.dssutl.out",IfNotFound=Ignore)
# The generated time series has identifier parts that indicate the HEC-DSS
# parts, other than the D part, which is related to the period of record.
SetOutputPeriod(OutputStart="1949-10-01",OutputEnd="1950-03-05")
NewPatternTimeSeries(Alias="Alias1",NewTSID="Apart1:Bpart1-with-dashes..Cpart1.Day.Fpart1",Description="Test data",SetStart="1950-01-01",SetEnd="1950-03-12",Units="CFS",PatternValues="5,10,12,13,75")
NewPatternTimeSeries(Alias="Alias2",NewTSID="Apart2-with-dashes:Bpart2..Cpart2.Day.Fpart2",Description="Test data",SetStart="1950-01-01",SetEnd="1950-03-12",Units="CFS",PatternValues="10,20,24,26,150")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_Day_GlobalOutputPeriod_out.dss",Type=PER-AVER)
#
# Use DSSUTL to output the data from the resulting DSS file to compare results
RunDSSUTL(DssFile="Results\Test_WriteHecDss_Day_GlobalOutputPeriod_out.dss",InputFile="Test_WriteHecDss_Day_GlobalOutputPeriod.MCO",DssutlProgram="${WorkingDir}\DSSUTL.EXE")
# In order to regenerate the expected results file, edit the output file in the file: Test_WriteHecDss_Day_GlobalOutputPeriod.MCO
CompareFiles(InputFile1="ExpectedResults\Test_WriteHecDss_Day_GlobalOutputPeriod.dssutl.out",InputFile2="Results/Test_WriteHecDss_Day_GlobalOutputPeriod.dssutl.out",IfDifferent=Warn)
