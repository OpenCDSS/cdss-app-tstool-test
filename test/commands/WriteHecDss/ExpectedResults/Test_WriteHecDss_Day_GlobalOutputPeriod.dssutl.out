  
 /APART1/BPART1-WITH-DASHES/CPART1/01JAN1950/1DAY/FPART1/
 Start: 01JAN1950 at 2400 hours;   End: 05MAR1950 at 2400 hours;  Number:   64
 Units: CFS        Type: PER-AVER
 01JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 08JAN1950, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 15JAN1950, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 22JAN1950, 2400;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 29JAN1950, 2400;    13.00   75.00    5.00   10.00   12.00   13.00   75.00
 05FEB1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 12FEB1950, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 19FEB1950, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 26FEB1950, 2400;    10.00   12.00   13.00   75.00    5.00   10.00   12.00
 05MAR1950, 2400;    13.00
 ----------------------------------------------------------------------------
  
 /APART2-WITH-DASHES/BPART2/CPART2/01JAN1950/1DAY/FPART2/
 Start: 01JAN1950 at 2400 hours;   End: 05MAR1950 at 2400 hours;  Number:   64
 Units: CFS        Type: PER-AVER
 01JAN1950, 2400;    10.00   20.00   24.00   26.00  150.00   10.00   20.00
 08JAN1950, 2400;    24.00   26.00  150.00   10.00   20.00   24.00   26.00
 15JAN1950, 2400;   150.00   10.00   20.00   24.00   26.00  150.00   10.00
 22JAN1950, 2400;    20.00   24.00   26.00  150.00   10.00   20.00   24.00
 29JAN1950, 2400;    26.00  150.00   10.00   20.00   24.00   26.00  150.00
 05FEB1950, 2400;    10.00   20.00   24.00   26.00  150.00   10.00   20.00
 12FEB1950, 2400;    24.00   26.00  150.00   10.00   20.00   24.00   26.00
 19FEB1950, 2400;   150.00   10.00   20.00   24.00   26.00  150.00   10.00
 26FEB1950, 2400;    20.00   24.00   26.00  150.00   10.00   20.00   24.00
 05MAR1950, 2400;    26.00
 ----------------------------------------------------------------------------
  
